src/TLuaInterpreter.cpp

```diff
@@ -6625,6 +6627,25 @@ int TLuaInterpreter::getCharacterName(lua_State* L)
     return 1;
 }
 
+// Documentation: https://wiki.mudlet.org/w/Manual:Miscellaneous_Functions#getProfileInformation
+int TLuaInterpreter::getProfileInformation(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    QString info = host.readProfileData(qsl("description"));
+    lua_pushstring(L, info.toUtf8().constData());
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Miscellaneous_Functions#setProfileInformation
+int TLuaInterpreter::setProfileInformation(lua_State* L)
+{
+    Host& host = getHostFromLua(L);
+    const QString text = getVerifiedString(L, __func__, 1, "text");
```

Comment:
:bulb: Might it be worth also accepting a `nil`as the argument - so that calling `setProfileInformation()` without ANY argument also does the clearing operation?

Reply:
added, also added that it revert to the default text immediately if nil or empty string supplied

Reply:
Sorry, missed this initially - but in Mudlet API design we use a 'clear*' function whenever we need to clear something instead of setting something to "" or nil. Or try to, anyway:

![image](https://github.com/user-attachments/assets/186c61cb-5281-4f9f-85a3-d950392f6bcf)

Reply:
ok, I can add that
but `setProfileInformation` should only accept a string then and not the nil, correct?

Reply:
Right

Reply:
:information_source: There are 12 `clear...(...)` functions AND 8 `reset...(...)` but it is not so easy to determine how many of the 103 `set...(...)` functions might have a clear/reset functionality...

Reply:
implemented `clearProfileInformation()` function
`setProfileInformation()` now only accepts a string, return nil and error message if string is empty
updated Area 51